soc/riscv/esp32c3/power.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/pm/pm.h>
+#include <zephyr/irq.h>
+#include <esp_sleep.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+/* Invoke Low Power/System Off specific Tasks */
+__weak void pm_state_set(enum pm_state state, uint8_t substate_id)
+{
+	ARG_UNUSED(substate_id);
+
+	switch (state) {
+	case PM_STATE_SOFT_OFF:
+		/* Forces RTC domain to be always on */
+		esp_sleep_pd_config(ESP_PD_DOMAIN_RTC_PERIPH, ESP_PD_OPTION_ON);
+		esp_deep_sleep_start();
+		break;
+	case PM_STATE_STANDBY:
+		/* Nothing to do. */
+		break;
+	default:
+		LOG_DBG("Unsupported power state %u", state);
+		break;
+	}
+}
+
+/* Handle SOC specific activity after Low Power Mode Exit */
+__weak void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
+{
+	ARG_UNUSED(substate_id);
+
+	switch (state) {
+	case PM_STATE_SOFT_OFF:
+		/* Nothing to do. */
+		break;
+	case PM_STATE_STANDBY:
+		irq_unlock(MSTATUS_IEN);
+		__asm__ volatile("wfi");
+		esp_light_sleep_start();
```

Comment:
@glaubermaroto
This is a very strange hack. Is there a reason you didn't put light sleep in a separate SUSPEND_TO_IDLE state, but enable it directly after waking from standby?

I'm trying to [refactor the interrupt enable handling across all architectures in the context of System PM wake-up](https://github.com/zephyrproject-rtos/zephyr/pull/60998#discussion_r1339312370), and this section is blocking that.

I'll gladly create a PR for moving this logic into its own PM_STATE, provided there isn't an explicit reason for this.